.github/actions/cache-node-modules/action.yml

```diff
@@ -0,0 +1,31 @@
+name: Cache node_modules and dist
+description: Caches the node_modules and optionally dist.
+inputs:
+    shared-dist:
+        description: Whether to cache shared/dist too.
+        default: 'true'
```

Comment:
I don't like the name of this action. The statement `uses: ./.github/actions/cache-node-modules` does not include the information that `shared/dist` is also cached. I'd suggest calling the action `cache` or `dfm-cache` and adding a second input for `node-modules` which also defaults to true.